lib/kernel/test/pg_SUITE.erl

```diff
@@ -620,13 +620,55 @@ group_leave(Config) when is_list(Config) ->
     1 = erlang:trace(PgPid, true, ['receive']),
     peer:stop(Peer),
     receive
-        {trace, PgPid, 'receive', {nodedown, Node}} -> ok
+        {trace, PgPid, 'receive', {nodedown, Node, _}} -> ok
     end,
     1 = erlang:trace(PgPid, false, ['receive']),
     sync(?FUNCTION_NAME),
     ?assertEqual([], pg:get_members(?FUNCTION_NAME, two)),
     ok.
 
+node_type(Config) when is_list(Config) ->
+    NodeTypeDefault = ?FUNCTION_NAME,
+    NodeTypeVisible = node_type_visible,
+    NodeTypeHidden = node_type_hidden,
+    NodeTypeAll = node_type_all,
+    Scopes = [NodeTypeDefault, NodeTypeVisible, NodeTypeHidden, NodeTypeAll],
+
+    {ok, _} = pg:start(NodeTypeVisible, #{node_type => visible}),
+    {ok, _} = pg:start(NodeTypeHidden, #{node_type => hidden}),
+    {ok, _} = pg:start(NodeTypeAll, #{node_type => all}),
+
+    Pid = erlang:spawn(forever()),
+    [?assertEqual(ok, pg:join(Scope, key, Pid)) || Scope <- Scopes],
+
+    {Peer, Node} = spawn_disconnected_node(NodeTypeDefault, node_type_visible, Config, []),
+    {ok, _} = peer:call(Peer, pg, start, [NodeTypeVisible, #{node_type => visible}]),
+    {ok, _} = peer:call(Peer, pg, start, [NodeTypeHidden, #{node_type => hidden}]),
+    {ok, _} = peer:call(Peer, pg, start, [NodeTypeAll, #{node_type => all}]),
+    true = net_kernel:connect_node(Node),
+
+    ?assertEqual([Pid], rpc:call(Node, pg, get_members, [NodeTypeDefault, key])),
```

Comment:
The test flakes here:

```
=== Location: [{pg_SUITE,node_type,650},
              {test_server,ts_tc,1782},
              {test_server,run_test_case_eval1,1291},
              {test_server,run_test_case_eval,1223}]
=== === Reason: {assertEqual,
                     [{module,pg_SUITE},
                      {line,650},
                      {expression,
                          "rpc : call ( Node , pg , get_members , [ NodeTypeDefault , key ] )"},
                      {expected,[<0.119765.0>]},
                      {value,[]}]}
  in function  pg_SUITE:node_type/1 (pg_SUITE.erl, line 650)
```

Because all operations are asynchronous, and by the time test runner reaches (and proceeds through) `rpc:call`, scopes haven't finished scope discovery.

If you look into other test cases, there are instructions like this:
```
    sync({?FUNCTION_NAME, Node}),
    sync(?FUNCTION_NAME),
```
Which ensure that a scope named `?FUNCTION_NAME` on the `Node` has processed the message queue.

Something like this should help:
`[begin sync(Proc, Node), sync(Proc) end || Proc <- [NodeTypeDefault, NodeTypeVisible, NodeTypeHidden, NodeTypeAll]],